modules/rhcos-load-firmware-blobs.adoc

```diff
@@ -0,0 +1,61 @@
+// Module included in the following assemblies:
+//
+// * post_installation_configuration/machine-configuration-tasks.adoc
+
+[id="rhcos-load-firmware-blobs_{context}"]
+
+= Loading custom firmware blobs in the machine config manifest
+
+Because the default location for firmware blobs in `/usr/lib` is read-only, you can locate a custom firmware blob by updating the search path. This enables you to load local firmware blobs in the machine config manifest when the blobs are not managed by {op-system}.
+
+.Procedure
+
+. Create a Butane config file, `98-worker-firmware-blob.bu`, that updates the search path so that it is root-owned and writable to local storage. The following example places the custom blob file under `/var/lib/firmware`.
++
+[NOTE]
+====
+See "Creating machine configs with Butane" for information about Butane.
+====
+.Butane config file for custom firmware blob
++
+[source,yaml]
+----
+variant: openshift
+version: 4.9.0
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: worker
+  name: 98-worker-firmware-blob
+storage:
+  files:
+  - path: /FWPATH/<package_name> <1>
+    mode: 0644 <2>
```

Comment:
As this is written, I don't think this is going to work for users. They also need an additional bit of configuration to write out the actual content of the file.  Using the example [MC in a gist](https://gist.githubusercontent.com/borball/c4061d23d06b2ee6eaf6cc87781045db/raw/3a951daec36e33d775597f48698a1be75a44df7b/mc-ice-ddp.yaml) from BZ#2010186, the more complete example would look something like:

```yaml
storage:
  files:
    - path: $FWPATH/<package_name>
      mode: 044
      contents: <base64 encoded contents of the file>
```

Reply:
Or maybe suggest the use of `--files-dir` here with `butane` so that you can generate the machine config using a local copy of the firmware blob?

See https://coreos.github.io/butane/examples/#files

Reply:
I've attempted to use your latter suggestion, adding the local storage file at line 33 and indicating in the callout note that the `--files-dir` option must be used with Butane. Additionally, I added the option to the butane command example on line 49. Please verify if this is what you had in mind or let me know if I missed the mark.